lib/internal/mime.js

```diff
@@ -0,0 +1,403 @@
+      const insideMatch = RegExpPrototypeExec(
+        QUOTED_VALUE_PATTERN,
+        StringPrototypeSlice(str, position));
+      position += insideMatch[0].length;
+      // Skip including last character if an unmatched '\' or '"' during
+      // unescape
+      const inside = insideMatch[1] || insideMatch[2] ?
+        StringPrototypeSlice(insideMatch[0], 0, -1) :
+        insideMatch[0];
+      // Unescape '\' quoted characters
+      parameterValue = removeBackslashes(inside);
+      // If we did have an unmatched '\' add it back to the end
+      if (insideMatch[1]) parameterValue += '\\';
+    } else {
+      // Handle the normal parameter value form
+      const valueEnd = StringPrototypeIndexOf(str, SEMICOLON, position);
+      const rawValue = valueEnd === -1 ?
+        StringPrototypeSlice(str, position) :
+        StringPrototypeSlice(str, position, valueEnd);
+      position += rawValue.length;
+      const trimmedValue = StringPrototypeSlice(
+        rawValue,
+        0,
+        SafeStringPrototypeSearch(rawValue, START_ENDING_WHITESPACE));
+      // Ignore parameters without values
+      if (trimmedValue === '') continue;
+      parameterValue = trimmedValue;
+    }
+    if (parameterString !== '' &&
+      SafeStringPrototypeSearch(parameterString,
+                                NotHTTPTokenCodePoint) === -1 &&
+      SafeStringPrototypeSearch(parameterValue,
+                                NotHTTPQuotedStringCodePoint) === -1 &&
+      paramsMap.has(parameterString) === false) {
+      paramsMap.set(parameterString, parameterValue);
+    }
+    position++;
+  }
+  return paramsMap;
+}
+
+function escapeQuoteOrSolidus(str) {
+  let ret = '';
+  for (const c of str) {
+    if (c === '"' || c === '\\') {
+      ret += `\\${c}`;
+    } else {
+      ret += c;
+    }
+  }
+  return ret;
+}
+
+const encode = (value) => {
+  if (value.length === 0) return '""';
+  NotHTTPTokenCodePoint.lastIndex = 0;
+  const encode = SafeStringPrototypeSearch(value, NotHTTPTokenCodePoint) !== -1;
+  if (!encode) return value;
+  const escaped = escapeQuoteOrSolidus(value);
+  return `"${escaped}"`;
+};
+
+const MIMEStringify = (type, subtype, parameters) => {
+  let ret = `${type}/${subtype}`;
+  const paramStr = MIMEParamsStringify(parameters);
+  if (paramStr.length) ret += `;${paramStr}`;
+  return ret;
+};
+
+const MIMEParamsStringify = (parameters) => {
+  let ret = '';
+  const entries = MIMEParamsData(parameters).entries();
+  let keyValuePair, done;
+  // Using this to avoid prototype pollution on Map iterators
+  // eslint-disable-next-line no-cond-assign
+  while ({ value: keyValuePair, done } = entries.next()) {
+    if (done) break;
+    const { 0: key, 1: value } = keyValuePair;
+    const encoded = encode(value);
+    // Ensure they are separated
+    if (ret.length) ret += ';';
+    ret += `${key}=${encoded}`;
+  }
+  return ret;
+};
+
+class MIMEParams {
+  #data;
+  constructor() {
+    this.#data = new Map();
```

Comment:
```suggestion
    this.#data = new SafeMap();
```

Reply:
I think line 21 of the file means that `Map` === `SafeMap` but maybe we shouldn't do that and keep `SafeMap` separate from `Map` to avoid confusion.